core/domain/question_services.py

```diff
@@ -199,10 +199,38 @@ def get_question_skill_links_of_skill(skill_id):
             question_skill_link_model.skill_id)
         for question_skill_link_model in question_skill_link_model_list
     ]
-
     return question_skill_links
 
 
+def delete_multi_question_skill_links(question_skill_links):
+    """Deletes multiple question skill links from the model.
+
+    Args:
+    question_skill_links: list(QuestionSkillLink). The list of
+        question skill link domain objects that need to be deleted
+        from the model.
+    """
+
+    for question_skill_link in question_skill_links:
```

Comment:
No, this still makes N calls to the datastore, each of which deletes a single entity. It's better to make a *_multi() call at the datastore layer, which makes just a single call that deletes multiple entities at once.

Search for delete_multi() in the codebase. I think you might be able to call it on the model class itself -- it's not the first time we've had to do this.

Similarly for put_multi().

Here's another reference: https://cloud.google.com/appengine/docs/standard/python/ndb/creating-entities#Python_Batch_operations. But it's better to use the methods we already have in the model classes than to call ndb directly.

Reply:
Ohh sorry I misunderstood your previous comment. Made those changes, I added put_multi and delete_multi functions to the model and am calling them from question_services